### PR TITLE
Add Auditlog tab to contact's detail page

### PR DESCRIPTION
**Pull request by virlatinus**
| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Related user documentation PR URL | N/A
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | N/A
| BC breaks? | N/A
| Deprecations? | N/A

#### Description:
This PR adds a new tab to the contact's details page that shows audit log entries for the contact. This will permit to debug changes to a contact (i.e. by integrations).

#### Steps to test this PR:
1. Apply this PR
2. Go to a contact's details page
3. You will see a new tab titled Audit log.
4. Click on it and it will show a table with 3 columns: User/Source Name (who/what updated the contact), Event Type and Timestamp.

### CODE REVIEW

File: app/bundles/LeadBundle/Entity/LeadEventLogRepository.php [hunk 1]
```diff
@@ -110,10 +110,10 @@ public function getEventsByLead($bundle, $object, Lead $lead, array $options)
             ->setParameter('object', $object);
 
         if (isset($options['search']) && $options['search']) {
-            $qb->andWhere($qb->expr()->like($alias.'.original_file', $qb->expr()->literal('%'.$options['search'].'%')));
+            $qb->andWhere($qb->expr()->like($alias.'.action', $qb->expr()->literal('%'.$options['search'].'%')));
```
Comment (alanhartless) [inline app/bundles/LeadBundle/Entity/LeadEventLogRepository.php]: I think this line and 113 needs to be reverted to latest staging where `properties` is used on line 113 and `original_file` is used on line 116. Unless this change affects the audit log then we need to discuss why the change is needed.

File: app/bundles/LeadBundle/Views/Auditlog/details.html.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+$text    = '';
+switch ($type) {
+    case 'create':
+        $text = 'The contact was created.';
+        if (isset($details['ipAddresses']) && is_array($details['ipAddresses'])) {
+            $text .= ' Origin IP: '.implode(', ', array_splice($details['ipAddresses'], 1));
+        }
+        break;
+    case 'delete':
+        $text = 'The contact was deleted.';
```
Comment (alanhartless) [inline app/bundles/LeadBundle/Views/Auditlog/details.html.php]: All these hard coded strings in English need to be converted to translation strings.

File: app/bundles/PluginBundle/Command/FetchLeadsCommand.php [hunk 3]
```diff
@@ -99,6 +99,16 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $integrationHelper = $container->get('mautic.helper.integration');
 
             $integrationObject = $integrationHelper->getIntegrationObject($integration);
+
+            if (!$integrationObject->isAuthorized()) {
+                $output->writeln(sprintf('<error>ERROR:</error> <info>%s is not authorized</info>', $integration));
```
Comment (alanhartless) [inline app/bundles/PluginBundle/Command/FetchLeadsCommand.php]: Let's use a translated string here too.

**Comment by mqueme** [comment]
Works. shows new tab with audit log

**Comment by luizeof** [comment]
Hello @virlatinus .. any screen ? :)

**Comment by virlatinus** [comment]
@luizeof 
image

**Comment by mqueme** [comment]
changes on this look so  much better. and info shows correctly.
